src/dataframerow/dataframerow.jl

```diff
@@ -455,55 +455,78 @@ Base.merge(a::DataFrameRow, b::DataFrameRow) = merge(NamedTuple(a), NamedTuple(b
 Base.merge(a::DataFrameRow, b::Base.Iterators.Pairs) = merge(NamedTuple(a), b)
 Base.merge(a::DataFrameRow, itr) = merge(NamedTuple(a), itr)
 
-# hash of DataFrame rows based on its values
-# so that duplicate rows would have the same hash
-# table columns are passed as a tuple of vectors to ensure type specialization
-rowhash(cols::Tuple{AbstractVector}, r::Int, h::UInt = zero(UInt))::UInt =
-    hash(cols[1][r], h)
-function rowhash(cols::Tuple{Vararg{AbstractVector}}, r::Int, h::UInt = zero(UInt))::UInt
-    h = hash(cols[1][r], h)
-    rowhash(Base.tail(cols), r, h)
-end
-
-Base.hash(r::DataFrameRow, h::UInt = zero(UInt)) =
-    rowhash(ntuple(col -> parent(r)[!, parentcols(index(r), col)], length(r)), row(r), h)
+Base.hash(r::DataFrameRow, h::UInt) = _nt_like_hash(r, h)
+
+for fun in (Base.isequal, Base.:(==))
+    function $fun(r1::DataFrameRow, r2::DataFrameRow)
+        if parent(r1) === parent(r2)
+            parentcols(index(r1)) == parentcols(index(r2)) || return false
+        else
+            _names(r1) == _names(r2) || return false
+        end
+        return all(((a, b),) -> $fun(a, b), zip(r1, r2))
+    end
 
-function Base.:(==)(r1::DataFrameRow, r2::DataFrameRow)
-    if parent(r1) === parent(r2)
-        parentcols(index(r1)) == parentcols(index(r2)) || return false
-        row(r1) == row(r2) && return true
-    else
-        _names(r1) == _names(r2) || return false
+    function $fun(r1::NamedTuple, r2::DataFrameRow)
+        propertynames(r1) == _names(r2) || return false
+        return all(((a, b),) -> $fun(a, b), zip(r1, r2))
     end
-    all(((a, b),) -> a == b, zip(r1, r2))
+
+    $fun(r1::DataFrameRow, r2::NamedTuple) = $fun(r2, r1)
 end
 
-function Base.isequal(r1::DataFrameRow, r2::DataFrameRow)
-    if parent(r1) === parent(r2)
-        parentcols(index(r1)) == parentcols(index(r2)) || return false
-        row(r1) == row(r2) && return true
-    else
-        _names(r1) == _names(r2) || return false
+
+for (fun1, fun2) in ((Base.isequal, Base.isless), (Base.:(==), Base.:(<)))
```

Comment:
Maybe call these `eqfun` and `cmpfun` or something explicit like that?

Reply:
OK - will change.